src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -245,6 +245,11 @@ private void AddOriginToResult(string origin, CorsPolicy policy, CorsResult resu
             else if (policy.Origins.Contains(origin))
             {
                 result.AllowedOrigin = origin;
+
+                if(policy.Origins.Count > 1)
```

Comment:
It appears even with single origin, this should be set
https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS
 
![image](https://cloud.githubusercontent.com/assets/6559784/21208117/da4eb43e-c220-11e6-851c-7b4927635745.png)

Reply:
@jbagga, was that page really updated? I looked through the history and could not find such a modification.

Interestingly,  the reference page for the `Access-Control-Allow-Origin` header still says that `vary: origin` must be set: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#CORS_and_caching

Perhaps, re-opening https://github.com/aspnet/CORS/issues/111 should be considered?

![image](https://user-images.githubusercontent.com/4699723/29855458-045063e4-8d1a-11e7-9f20-f27295f2e954.png)

Reply:
@nickwender - can you file a new issue for this? It's a bit hard to follow the discussion in this old PR.